appserver/payara-appserver-modules/microprofile/fault-tolerance/src/main/java/fish/payara/microprofile/faulttolerance/interceptors/fallback/FallbackPolicy.java

```diff
@@ -66,18 +67,20 @@
     
     private static final Logger logger = Logger.getLogger(FallbackPolicy.class.getName());
     
-    private final Class<? extends FallbackHandler> fallbackClass;
+    private final Class<? extends FallbackHandler<?>> fallbackClass;
     private final String fallbackMethod;
     
+    @SuppressWarnings("unchecked")
     public FallbackPolicy(Fallback fallback, Config config, InvocationContext invocationContext) 
             throws ClassNotFoundException {     
-        fallbackClass = (Class<? extends FallbackHandler>) Thread.currentThread().getContextClassLoader().loadClass(
-                (String) FaultToleranceCdiUtils.getOverrideValue(config, Fallback.class, "value", 
-                        invocationContext, String.class)
-                .orElse(fallback.value().getName()));
-        
-        fallbackMethod = (String) FaultToleranceCdiUtils.getOverrideValue(config, Fallback.class, 
-                "fallbackMethod", invocationContext, String.class)
+        Optional<String> className = FaultToleranceCdiUtils.getOverrideValue(config, Fallback.class, "value", 
+                invocationContext, String.class);
+        fallbackClass = className.isPresent()
+                ? (Class<? extends FallbackHandler<?>>) Thread.currentThread().getContextClassLoader()
+                        .loadClass(className.get())
+                : fallback.value();
```

Comment:
Note similar case. Presumably config returns the class name as `String`. If it is present we load it, otherwise use the instance from the annotation. Here `map` cannot be used due to generic capture evaluation by the compiler. Also the conditional avoids calling `fallback.value()` in case override is present which could be relevant in avoiding loading the class.

Reply:
For this one I *think* yes it does read it as a String (Afaik anyway).